src/store/deployments/deploymentsReducer.js

```diff
@@ -4,22 +4,73 @@ import actionTypes from './actionTypes';
 // INITIAL STATE
 const initialState = [];
 
+/**
+ * deployments reducer
+ *
+ * @param state
+ * @param action
+ * @returns {Array} Deployments list
+ */
 const deploymentsReducer = (state = initialState, action = undefined) => {
   switch (action.type) {
-    case actionTypes.DELETE_DEPLOYED_EXPERIMENT:
-      return state.filter((deployment) => {
-        return deployment.experimentId !== action.experimentId;
-      });
-    case actionTypes.FETCH_DEPLOYED_EXPERIMENTS:
-      return [...action.deployments];
+    // reset initial fetch of deployments
+    case actionTypes.CLEAR_ALL_DEPLOYMENTS:
+      return [];
+    
+    // SUCCESS
+    // fetch deployments success
     case actionTypes.FETCH_DEPLOYMENTS_SUCCESS:
       return [...action.deployments];
+    // create deployment success
+    case actionTypes.CREATE_DEPLOYMENT_SUCCESS:
+      return [...state, action.deployment];
+    // update deployment success
+    case actionTypes.UPDATE_DEPLOYMENT_SUCCESS:
+      return [...action.deployments];
+    // delete deployment success
+    case actionTypes.DELETE_DEPLOYMENT_SUCCESS:
+      return [...action.deployments];
+    
+    // // // // // // //
+
+    // FAIL
+    // fetch deployments fail
     case actionTypes.FETCH_DEPLOYMENTS_FAIL:
-      return [];
+      return [...state];
+    // create deployment fail
+    case actionTypes.CREATE_DEPLOYMENT_FAIL:
+      return [...state];
+    // update deployment fail
+    case actionTypes.deploymentLOYMENT_FAIL:
+      return [...state];
+    // delete deployment fail
+    case actionTypes.DELETE_DEPLOYMENT_FAIL:
+      return [...state];
+    
+    // // // // // // //
+
+    // DEFAULT
     default:
       return state;
   }
 };
 
+// SELECTOR
+
+/**
+ * Select deployment by id
+ *
+ * @param {Array} state
+ * @param {string} deploymentId Deployment UUID
+ * @returns {Array}
+ */
+export const getDeploymentById = (state, deploymentId) => {
+  return state.filter((deployment) => {
+    if (deployment.uuid == deploymentId) {
+      return deployment
+    }
+  });
```

Comment:
@miguelfferraz deu estes warnings aqui:
 Line 81:55:  Expected to return a value at the end of arrow function  array-callback-return
  Line 82:25:  Expected '===' and instead saw '=='                      eqeqeq